cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -67,44 +67,26 @@ constexpr Unsigned to_unsigned(T signed_) {
   return static_cast<Unsigned>(signed_);
 }
 
+template <bool Checked>
 struct AbsoluteValue {
   template <typename T, typename Arg>
-  static constexpr enable_if_floating_point<T> Call(KernelContext*, T arg, Status*) {
-    return std::fabs(arg);
-  }
-
-  template <typename T, typename Arg>
-  static constexpr enable_if_unsigned_integer<T> Call(KernelContext*, T arg, Status*) {
-    return arg;
-  }
-
-  template <typename T, typename Arg>
-  static constexpr enable_if_signed_integer<T> Call(KernelContext*, T arg, Status* st) {
-    return (arg < 0) ? arrow::internal::SafeSignedNegate(arg) : arg;
-  }
-};
-
-struct AbsoluteValueChecked {
-  template <typename T, typename Arg>
-  static enable_if_signed_integer<T> Call(KernelContext*, Arg arg, Status* st) {
-    static_assert(std::is_same<T, Arg>::value, "");
-    if (arg == std::numeric_limits<Arg>::min()) {
-      *st = Status::Invalid("overflow");
+  static constexpr T Call(KernelContext*, T arg, Status* st) {
+    if constexpr (std::is_floating_point<T>::value) {
+      return std::fabs(arg);
+    } else if constexpr (std::is_signed<T>::value) {
```

Comment:
cpplint is failing for some of these: https://github.com/apache/arrow/pull/10414/checks?check_run_id=2687678079#step:7:2040

It looks like this has been fixed upstream: https://github.com/cpplint/cpplint/pull/136